src/server.c

```diff
@@ -2933,9 +2933,9 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
      * our clients. */
     updateFailoverStatus();
 
-    /* Flush the pending invalidation messages to clients participating to the
-     * client side caching protocol in general mode */
-    trackingHandlePendingKeyInvalidations();
+    /* We can't have pending invalidation messages to clients participating
+     * to the client side caching protocol in general mode */
+    serverAssert(listLength(server.tracking_pending_keys) == 0);
```

Comment:
```suggestion
    /* Since we rely on current_client to send scheduled invalidation messages
     * we have to flush them after each command, so when we get here, the list
     * must be empty. */
    serverAssert(listLength(server.tracking_pending_keys) == 0);
```